pulp_ansible/app/global_access_conditions.py

```diff
@@ -0,0 +1,38 @@
+from pulp_ansible.app import models
+
+
+def v3_can_view_repo_content(request, view, action):
+    """
+    Check if the repo is private, only let users with view repository permissions
+    view the collections here.
+    """
+
+    if "distro_base_path" in view.kwargs:
+        distro_base_path = view.kwargs["distro_base_path"]
+        repo = models.AnsibleDistribution.objects.get(base_path=distro_base_path).repository.cast()
```

Comment:
This doesn't handle the scenario where the distribution is pointing at a repository version, or nothing at all!
```suggestion
        distro_base_path = view.kwargs["distro_base_path"]
        distro = models.AnsibleDistribution.objects.select_related("repository", "repository_version").get(base_path=distro_base_path)
        if not repo := distro.repository:
            if not repo_ver := distro.repository_version:
                 return False
            repo = repo_ver.repository
```